src/chains/evm/hub/modules/folks-hub-loan.ts

```diff
@@ -495,13 +508,20 @@ export function getUserLoansInfo(
 }
 
 export function simulateLoanChanges(loan: LoanManagerUserLoan, changes: Array<LoanChange>): LoanManagerUserLoan {
-  const [accountId, loanType, oldColPools, oldBorPools, oldCollaterals, oldBorrows] = loan;
+  const {
+    accountId,
+    loanTypeId,
+    colPools: oldColPools,
+    borPools: oldBorPools,
+    userLoanCollateral: oldCollaterals,
+    userLoanBorrow: oldBorrows,
+  } = loan;
 
   // make copy which can update
   const colPools = [...structuredClone(oldColPools)];
   const borPools = [...structuredClone(oldBorPools)];
-  const collaterals = [...structuredClone(oldCollaterals)];
-  const borrows = [...structuredClone(oldBorrows)];
+  const userLoanCollateral = [...structuredClone(oldCollaterals)];
+  const userLoanBorrow = [...structuredClone(oldBorrows)];
```

Comment:
I don't think there is a need for `[...]` syntax anymore since the type is no longer `readonly`

Reply:
yep, thanks 5b8796a53efe2a8d461b455ac8e35c6f2ddcccd3

Reply:
@palace22 I think you still need to use structuredClone, but this way without the spread:
```ts
const colPools = structuredClone(oldColPools);
const borPools = structuredClone(oldBorPools);
const collaterals = structuredClone(oldCollaterals);
const borrows = structuredClone(oldBorrows);
```

Reply:
Here the reason why https://github.com/Folks-Finance/folks-finance-xchain-js-sdk/pull/85#discussion_r1669007502

Reply:
@stefanofa thanks updated, d52a73612172001cb99f38fe698751f7e7a9ee20

Reply:
Maybe it could be even better to do:
```ts
const {
  accountId,
  loanTypeId,
  colPools,
  borPools,
  userLoanCollateral,
  userLoanBorrow,
} = structuredClone(loan);
```
wdyt?